### PR TITLE
feature: Internal build mode

### LINKED ISSUE #476: Internal build mode
## Use Case

Add a global switch for internal vs. public-facing builds. This is to allow scientists to access experimental features when working on new datasets.

## Acceptance Criteria

_Please describe how you know this is done_

## Details

_Please provide any helpful specifications_

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
Closes #476, adding an internal build mode.

*Estimated review size: small, 5 minutes*

Solution
========
- Adds a second internal build config for `vite` which can be accessed by running `npm run start-internal` or `npm run build-internal`.
- Adds a second PR preview that builds the internal version. (These can be accessed by replacing the `pr-preview` folder in the URL with `pr-preview-internal`.)
- Adds a global constant that reads whether the app is in internal mode.
- Added indicators for internal mode to the header, version info, and console log.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open internal preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-477/
2. Go to the help menu and open "Version Info". You should see text stating that this is an internal build.
3. Repeat the same steps with the regular preview link. The message should not appear. https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview/pr-477/

Screenshots (optional):
-----------------------

image

image

Keyfiles (delete if not relevant):
-----------------------
1. main file/entry point
4. other important file

Thanks for contributing!

### CODE REVIEW

File: src/components/Dropdowns/HelpDropdown.tsx
Comment (ShrimpCryptid) [inline src/components/Dropdowns/HelpDropdown.tsx:1]: Show in version info

File: vite.config.js
Comment (ShrimpCryptid) [inline vite.config.js:1]: Export existing config so it can be overridden/extended by the `vite.config.internal.js` config.

Comment (toloudis) [inline vite.config.js:1]: Why not move this file into the new config directory

**Comment by toloudis** [comment]
This change probably merits an update in CONTRIBUTING.md or README to clarify the new start variations.

File: vite.config.js
Comment (ShrimpCryptid) [reply vite.config.js:1]: It's in the default location that `vite` will look for first... I could move it but I'd need to update all the run commands. Worth it?

Comment (toloudis) [reply vite.config.js:1]: I would prefer to have both config files in same directory whether it's config or project root. I would be fine with bringing the internal one out to be where the default one is, also.

Comment (toloudis) [reply vite.config.js:1]: Another possibility that would make me feel better is to rename config to internal-build-config to be more explicit about what it is.

Comment (ShrimpCryptid) [reply vite.config.js:1]: Renamed to `vite.internal-build.config.js`!

**Review by meganrm — APPROVED** [review]

**Pull request merged**